tests/spec/README.md

```diff
@@ -0,0 +1,112 @@
+# Spec tests
+
+These tests specify the expected behaviour of ehrQL, and by extension, the QM.
+
+They are also used to generate documentation.
+
+
+## Table of Contents
+
+`toc.py` defines the Table of Contents for the documentation generated by tests. A new spec test
+subdirectory or file must be defined here in order for it to appear in the docs.
+
+## Folder structure
+
+Within the spec directory, each test directory is considered a chapter in the docs, each file
+is considered to be a section, and each test is a paragraph.
+
+e.g.
+```
+tests
+    ├── spec
+    │   ├── aggregate_frame                       <-- chapter
+    │   │   ├── __init__.py                       <-- contains chapter title
+    │   │   ├── test_count_for_patient.py         <-- section
+    │   │   └── test_exists_for_patient.py        <-- section
+```
+
+### Titles
+Chapter titles are specified with a `title` attribute the test directory's `__init__.py`.
+Section titles are specified with a `title` attribute in each test file.
+Paragraph titles are extracted from the test names.
```

Comment:
suggestion: Maybe worth a quick mention that the default paragraph title can be overridden by adding a `title` attribute to the test function?

Iain added this feature in #602.